### PR TITLE
Expression: Add test framework for vectorized expression evaluation functions

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve? 
Add an automated test framework for vectorized expression evaluation functions.

### What is changed and how it works?
1. Add function `genVecBuiltinFuncBenchCase` to generate builtinFunc for the next unit test and benchmark test
2. Add function `TestVectorizedBuiltinFunc` for unit test for builtin Function expression.
3. Add function `BenchmarkVectorizedBuiltinFunc` for benchmark test for builtin Function expression.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: tools/check/go.mod [hunk 1]
```diff
@@ -20,3 +20,5 @@ require (
 	gopkg.in/yaml.v2 v2.2.2 // indirect
 	honnef.co/go/tools v0.0.0-20180920025451-e3ad64cb4ed3
 )
+
+replace golang.org/x/tools => golang.org/x/tools v0.0.0-20190829051458-42f498d34c4d
```
Comment (zz-jason) [inline tools/check/go.mod]: why not directly update the version of `golang.org/x/tools`?

File: expression/bench_test.go [hunk 2]
```diff
@@ -255,7 +332,7 @@ func genVecExprBenchCase(ctx sessionctx.Context, testCase vecExprBenchCase) (Exp
 	return expr, input, output
 }
 
-func TestVectorizedExpression(t *testing.T) {
+func TestVectorizedEvalOneVec(t *testing.T) {
 	ctx := mock.NewContext()
 	for _, testCase := range vecExprBenchCases {
```
Comment (zz-jason) [inline expression/bench_test.go]: After this PR, contributors who reimplemented a new scalar function should update `vecExprBenchCases` to get that function benchmarked?

Comment (qw4990) [reply expression/bench_test.go]: Yes. 
`TestVectorizedBuiltinFunc` will read this list to do some correctness tests by comparing the row-based results and the vectorized results. 
`BenchmarkVectorizedBuiltinFunc` will use it to do some benchmarks.
And `TestVectorizedEvalOneVec` is the individual test for `evalOneColumn` and `evalOneVec`, which are used in projection operators.

**Comment by SunRunAway** [comment]
Please see https://help.github.com/en/articles/closing-issues-using-keywords, to automatically close issues in GitHub.

File: expression/bench_test.go (hunk 2 above)
Comment (zz-jason) [reply expression/bench_test.go]: If each PR needs to modify this list, I think the merge conflict between PRs would be very heavy.

Comment (qw4990) [reply expression/bench_test.go]: But these conflicts are small(one line).
Would you have any other suggestions?

Comment (qw4990) [reply expression/bench_test.go]: The `vecExprBenchCases` is split to multiple lists by function name.
PTAL.

File: tools/check/go.mod (hunk 1 above)
Comment (qw4990) [reply tools/check/go.mod]: updated, PTAL.

File: expression/bench_test.go [hunk 3]
```diff
@@ -229,12 +295,24 @@ func eType2FieldType(eType types.EvalType) *types.FieldType {
...
+		return types.NewFieldType(mysql.TypeVarString)
+	default:
+		panic(fmt.Sprintf("EvalType=%v is not supported.", eType))
 	}
-	// TODO: support all EvalTypes later.
-	panic(fmt.Sprintf("EvalType=%v is not supported.", eType))
 }
 
-func genVecExprBenchCase(ctx sessionctx.Context, testCase vecExprBenchCase) (Expression, *chunk.Chunk, *chunk.Chunk) {
+func genVecExprBenchCase(ctx sessionctx.Context, funcName string, testCase vecExprBenchCase) (Expression, *chunk.Chunk, *chunk.Chunk) {
```
[the remark this answers is not in the document]
Comment (qw4990) [reply expression/bench_test.go]: updated.

File: expression/bench_test.go [hunk 4]
```diff
@@ -255,58 +333,375 @@ func genVecExprBenchCase(ctx sessionctx.Context, testCase vecExprBenchCase) (Exp
 	return expr, input, output
 }
 
-func TestVectorizedExpression(t *testing.T) {
+func TestVectorizedEvalOneVec(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (qw4990) [reply expression/bench_test.go]: updated.

**Pull request merged**